supra-cli/src/lib.rs

```diff
@@ -0,0 +1,60 @@
+// use libp2p::PeerId;
+pub use sc_cli::{CliConfiguration, SharedParams};
+use structopt::StructOpt;
+use bs58::{decode, Alphabet};
+use std::str::FromStr;
+
+#[derive(Debug, StructOpt)]
+pub struct SupraCmd {
+    #[allow(missing_docs)]
+	#[structopt(flatten)]
+	pub shared_params: SharedParams,
+
+    #[allow(missing_docs)]
+    #[structopt(flatten)]
+    pub peer_id_hex: PeerIdHexCmd,
+}
+
+#[derive(Debug, StructOpt)]
+pub struct PeerIdHexCmd {
+    /// Provides Hex version of PeerId as base58.
+    #[structopt()]
+    pub peer_id: String,
+
+    #[allow(missing_docs)]
+	#[structopt(flatten)]
+	pub shared_params: SharedParams,
+}
+
+impl PeerIdHexCmd {
+    pub fn convert_to_hex(&self) -> Result<(), sc_cli::Error> {
+        println!("Converting Peer ID in base58 to hex");
+
+        let decoded_peer_id = decode(&self.peer_id)
+            .with_alphabet(Alphabet::BITCOIN)
+            .into_vec()
+            .unwrap();
+
+        println!("Decoded PeerId in bytes: {:?}", decoded_peer_id.clone());
```

Comment:
Instead of creating another variable, I think we can directly do - `println!("Decoded PeerId in hex: {:?}", hex::encode(decoded_peer_id)`